### PR TITLE
Designs

### PR DESCRIPTION
**Pull request by annasirunian**
1. Designs table (todo: restyle columns in expanded rows)
2. Shared Deletion Component

### CODE REVIEW

File: src/components/shared/NewProject.vue [hunk 1]
```diff
@@ -80,6 +80,7 @@ export default Vue.extend({
   methods: {
     createProject() {
       this.$store.commit("toggleDialog", "loader");
+      const a = this.projectName.value;
```
Comment (kvikshaug) [inline src/components/shared/NewProject.vue]: Leftover from debugging?

File: src/components/shared/DeletionDialog.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      axios
+        .all(
+          this.items.map(item =>
+            axios.delete(
+              `${settings.apis[storage]}/${this.itemsType}/${item.id}`
+            )
+          )
+        )
+        .then(
+          axios.spread(aaa => {
```
Comment (kvikshaug) [inline src/components/shared/DeletionDialog.vue]: This should probably be called `acct` to match the parameter, or just omitted if not in use

Comment (annasirunian) [reply src/components/shared/DeletionDialog.vue]: yes, thanks, will fix that

File: src/components/Designs.vue [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        isReactionKnockin && method === "Pathway"
+          ? JSON.parse(reaction).bigg_id
+          : reaction;
+      if (reactionId.startsWith("MNX")) {
+        return `https://www.metanetx.org/equa_info/${reactionId}`;
+      }
+      return `http://bigg.ucsd.edu/search?query=${reactionId}`;
+    },
+    geneLink(geneId) {
+      return `http://bigg.ucsd.edu/search?query=${geneId}`;
```
Comment (kvikshaug) [inline src/components/Designs.vue]: Not a big deal but I don't really feel this needs to be a method, just put the URL directly in the `href`.

File: src/components/shared/DeletionDialog.vue [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      const storage = {
+        designs: "designStorage",
+        maps: "maps",
+        models: "modelsStorage",
+        projects: "iam"
+      }[this.itemsType];
+
+      const ids = this.items.map(item => item.id);
+      axios
+        .all(
```
Comment (kvikshaug) [inline src/components/shared/DeletionDialog.vue]: Nice use of concurrent requests :+1:

File: src/components/Designs.vue [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+  name: "Designs",
+  data: () => ({
+    selected: [],
+    expand: true,
+    showAllReactionKnockins: false,
+    showAllReactionKnockouts: false,
+    showAllGeneKnockouts: false,
+    isDeletionDialogVisible: false,
+    isLoading: false,
+    headers: [
```
Comment (Midnighter) [inline src/components/Designs.vue]: Your column widths only add up to 95%, could that be a reason for layout problems when you expand?

Comment (annasirunian) [reply src/components/Designs.vue]: Oh, thank you, I should change it to 100%, initially I left 5% for checkbox area, but I couldn't specify width in the same way, expanded rows still need restyling though

**Pull request merged**